### PR TITLE
[modbus] Added support for RTU encoding over TCP

### PR DESCRIPTION
**Pull request by andrewfg**
Until now, this binding did not did not support RTU encoding over TCP.

This PR adds support for RTU encoding over TCP.

To be specific, the existing the existing `tcp` Thing has an additional Configuration Parameter `rtuEncoded` that allows selection between the existing 'TCP over TCP' transport and the new 'RTU over TCP' transport. If not specifically defined, the value of `rtuEncoded` defaults to `false` to ensure that there is no effect on existing users.

NOTE: this PR requires the following PRs..
- `jamod` [modbus] add support for RTU encoding over TCP https://github.com/openhab/jamod/pull/8
- `org.openhab.binding.modbus` [modbus] add support for RTU encoding over TCP https://github.com/openhab/openhab-core/pull/1965

### CODE REVIEW

File: bundles/org.openhab.binding.modbus/README.md [hunk 1]
```diff
@@ -37,8 +37,8 @@ The binding can also *write* data to Modbus slaves using FC05 (Write single coil
 
 Please note the following caveats or limitations
 
-* the binding does *not* act as Modbus slave (e.g. as Modbus TCP server).
-* the binding does *not* support Modbus RTU over Modbus TCP, also known as "Modbus over TCP/IP" or "Modbus over TCP" or "Modbus RTU/IP", although normal "Modbus TCP" is supported. However, there is a workaround: you can use a Virtual Serial Port Server, to emulate a COM Port and Bind it with openHAB using Modbus Serial.
+* The binding does *not* act as Modbus slave (e.g. as Modbus TCP server).
+* The binding *does* support Modbus RTU over Modbus TCP, (also known as "Modbus over TCP/IP" or "Modbus over TCP" or "Modbus RTU/IP"), as well as normal "Modbus TCP". Alternatively, you can use a Virtual Serial Port Server, to emulate a COM Port and Bind it with openHAB using Modbus Serial.
```
Comment (ssalonen) [inline bundles/org.openhab.binding.modbus/README.md:41]: Can we drop the comment on virtual port? Would there be any reason to prefer that over the built-in support?

Comment (andrewfg) [reply bundles/org.openhab.binding.modbus/README.md]: Ok. Done.

**Comment by ssalonen** [comment]
@giovannimirulla can you please  review the modbus.studer part?

I adviced @andrewfg to add support for the tcp thing in studer binding as well. This should enable support for users with serial/ethernet gateways.

**Comment by t2000** [comment]
Since https://github.com/openhab/openhab-core/pull/1965 has been merged we should finish this PR soon, or at least fix the compilation issue:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project org.openhab.binding.modbus: Compilation failure
[ERROR] /home/jenkins/jenkins-agent1/workspace/PR-openHAB-Addons/bundles/org.openhab.binding.modbus/src/main/java/org/openhab/binding/modbus/internal/handler/ModbusTcpThingHandler.java:[54,20] The constructor org.openhab.core.io.transport.modbus.endpoint.ModbusTCPSlaveEndpoint(java.lang.String, int) is undefined
```

because otherwise all builds of the addons pipeline will fail.

**Review by cpmeister — APPROVED** [review]

**Comment by andrewfg** [comment]
Many thanks @cpmeister

**Pull request merged**